shadows/framework/src/main/java/org/robolectric/shadows/ImageUtil.java

```diff
@@ -102,16 +102,16 @@ public static boolean writeToStream(
         ImageWriteParam iwparam = writer.getDefaultWriteParam();
         iwparam.setCompressionMode(ImageWriteParam.MODE_EXPLICIT);
         iwparam.setCompressionQuality((quality / 100f));
+        int width = realBitmap.getWidth();
+        int height = realBitmap.getHeight();
         BufferedImage bufferedImage =
             new BufferedImage(
-                realBitmap.getWidth(),
-                realBitmap.getHeight(),
+                width,
+                height,
                 getBufferedImageType(realBitmap.getConfig(), needAlphaChannel(format)));
-        for (int x = 0; x < realBitmap.getWidth(); x++) {
-          for (int y = 0; y < realBitmap.getHeight(); y++) {
-            bufferedImage.setRGB(x, y, realBitmap.getPixel(x, y));
-          }
-        }
+        int[] pixels = new int[width * height];
```

Comment:
Instead of the extra `new int[width * height]` allocation, which may be many MB, can you add a package-private method named getPixelsInternal or getColorsData to ShadowBitmap to get the colors array? That way it can copy directly into the ShadowBitmap colors array.